### DISCUSSION TITLE
Where to store a pointer to a resource(such as db) for future reference?

### DISCUSSION THREAD

**Discussion by axshor** [pocketbase/pocketbase#5699] [Q&A]
In my case I need to use a k/v database to persist some data, I can setup the db in the App event, how can I make the db pointer(reference) is available in other middlewares.

**Comment by whmeitzler** [comment]
Can you clarify what handle you want to keep? A reference to the record, or the DB itself?

In general, you can either query the DB using `e.App().Dao()...` or use the in-memory cache (think Redis, but you can keep functions or structs in there) at `e.App().Store()...`

**Comment by dschissler** [comment]
You're going to need to wrap Pocketbase http handlers with your own struct that (most likely) embeds a PB `core.App`.  Then attach custom DB systems to that.  You won't have access to the DB directly from the events via the parameter, so you'll have to be obtaining it directly from the wrapped context.  If you use any transactions then it could get weird, so you'll probably only want to include these things in the post event hooks since those aren't cancelable.
